cachi2/core/package_managers/pip.py

```diff
@@ -273,74 +273,61 @@ def _generate_purl_dependency(package: dict[str, Any]) -> str:
         if pyproject_toml.check_dynamic_version():
-            log.warning("Parsing dynamic metadata from pyproject.toml is not supported")
+            log.warning("Dynamic version parsing from pyproject.toml is not supported")
 
-        name = pyproject_toml.get_name()
-        version = pyproject_toml.get_version()
+        name, version = pyproject_toml.get_name(), pyproject_toml.get_version()
+        if name and version:
+            return name, version
 
-    if None in (name, version) and setup_py.exists():
-        log.info("Filling in missing metadata from setup.py")
-        name = name or setup_py.get_name()
-        version = version or setup_py.get_version()
+    # setup.py
+    if setup_py.exists():
+        log.debug("Checking setup.py for metadata")
+        name, version = setup_py.get_name(), setup_py.get_version()
+        if name and version:
+            return name, version
 
-    if None in (name, version) and setup_cfg.exists():
-        log.info("Filling in missing metadata from setup.cfg")
-        name = name or setup_cfg.get_name()
-        version = version or setup_cfg.get_version()
+    # setup.cfg
+    if setup_cfg.exists():
+        log.debug("Checking setup.cfg for metadata")
+        return setup_cfg.get_name(), setup_cfg.get_version()
 
-    if not name:
-        log.info("Processing metadata from git repository")
-        try:
-            repo_path = get_repo_id(package_dir.root).parsed_origin_url.path.removesuffix(".git")
-            repo_name = Path(repo_path).name
-            package_subpath = package_dir.subpath_from_root
-
-            resolved_path = Path(repo_name).joinpath(package_subpath)
-            normalized_path = canonicalize_name(str(resolved_path).replace("/", "-"))
-            name = normalized_path.strip("-.")
-        except UnsupportedFeature:
-            raise PackageRejected(
-                reason="Could not take name from the repository origin url",
-                solution=(
-                    "Please specify package metadata in a way that Cachi2 understands"
-                    " (see the docs)\n"
-                    "or make sure that the directory Cachi2 is processing is a git"
-                    " repository with\n"
-                    "an 'origin' remote in which case Cachi2 will infer the package name from"
-                    " the remote url."
-                ),
-                docs=PIP_METADATA_DOC,
-            )
+    return None, None
 
-    log.info("Resolved package name: %r", name)
-    if version:
-        log.info("Resolved package version: %r", version)
-    else:
-        log.warning("Could not resolve package version")
 
-    return name, version
+def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, Optional[str]]:
+    """Attempt to retrieve name and version of a pip package."""
+    name, version = _extract_metadata_from_config_files(package_dir)
+    return name or _infer_package_name_from_origin_url(package_dir), version
```

Comment:
I think having this conditionally assigned is more readable, but I believe this was @a-ovchinnikov review suggestion IIRC, so I'll refrain from demanding a conflicting approach, so let's keep it.

Reply:
I would prefer to keep the previous approach too.